system/iptables.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Linus Unnebäck <linus@folkdatorn.se>
+#
+# This file is part of Ansible
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+# import module snippets
+from ansible.module_utils.basic import *
+
+DOCUMENTATION = '''
+---
+module: iptables
+short_description: Modify the systems iptables
+requirements: []
+version_added: 1.9
+author:
+  - "Linus Unnebäck"
+description:
+  - Modify the systems iptables.
+options:
+  table:
+    description: Packet matching table to operate on.
+    required: false
+    default: filter
+    choices: [ "filter", "nat", "mangle", "raw", "security" ]
+  chain:
+    description: Chain to operate on.
+    required: true
+    choices: [ "INPUT", "FORWARD", "OUTPUT", "PREROUTING", "POSTROUTING", "SECMARK", "CONNSECMARK" ]
+  rule:
```

Comment:
That is one way of doing it, I started looking at it but there is **very many** options that we need to support in this case.

First of all we need all the basic standard options, that's not too bad. It would be ~11 options, but that wouldn't even cover my basic example of port forwarding which I think is fairly common. To make this work we would need to support the extended set of options which seems to be **very many**. Here is the list: http://ipset.netfilter.org/iptables-extensions.man.html

All rules becomes a bit longer but it looks kinda nice.

```diff
+ iptables: table=nat chain=PREROUTING in_interface=eth0 protocol=tcp match=tcp destination_port=80 jump=REDIRECT to_ports=8600'
- iptables: table=nat chain=PREROUTING rule='-i eth0 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 8600'
```

I don't know, what do you think?